ldap/servers/plugins/replication/cl5_clcache.c

```diff
@@ -467,17 +498,24 @@ clcache_load_buffer_bulk(CLC_Buffer *buf, int flag)
 
         if (use_flag == DB_NEXT) {
             /* For bulk read, position the cursor before read the next block */
-            rc = cursor->c_get(cursor,
-                               &buf->buf_key,
-                               &buf->buf_data,
-                               DB_SET);
+            rc = clcache_cursor_set(cursor, buf);
         }
 
-        /*
-         * Continue if the error is no-mem since we don't need to
-         * load in the key record anyway with DB_SET.
-         */
         if (0 == rc || DB_BUFFER_SMALL == rc) {
```

Comment:
I made a previous hotfix #4492 that stop replication session when detecting a big jump backward. The hotfix was failing because replication got broken with constant big jumps session after session. Without that hotfix replication kept going on even if restarted from the beginning of the CL.

So I prefer to get a permissive code that let replication going on even at a cost of a big jump.
In case of DB_BUFFER_SMALL during DB_SET (that I hope will no long happen) we just log a warning to help diagnose

Reply:
Fine. Anyway as I do not think that we could hit the double DB_BUFFER_SMALL, I have no strong feeling about the way we handle it ! -;)